### ISSUE TITLE
macports cmake failure

### ISSUE THREAD

**Issue by diwu1989** [FreeRDP/FreeRDP#2545]
On the latest version of macports, freerdp fails to compile because of cmake errors.

This is the cmake log: http://hastebin.com/kucobecepa.md

**Comment by bmiklautz** [comment]
@diwu1989 what version of FreeRDP are you trying to build and what cmake version are you using?

**Comment by bmiklautz** [comment]
@diwu1989 what exactly is the error you get? The cmake log doesn't say much.

**Comment by diwu1989** [comment]
FreeRDP version is FreeRDP-1.1.0-beta+2013071101
Cmake is version 3.2.1

This is the error log from cmake:
http://hastebin.com/wexoyeseru.md

This is the actual command that failed:
```
cmake -DCMAKE_INSTALL_PREFIX=/opt/local -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_INSTALL_RPATH=/opt/local/lib -DCMAKE_INSTALL_NAME_DIR=/opt/local/lib -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/usr" -DCMAKE_MODULE_PATH=/opt/local/share/cmake/Modules -DCMAKE_FIND_FRAMEWORK=LAST -Wno-dev -DWITH_X11=ON -DWITH_FFMPEG=ON -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.10" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_net_FreeRDP/FreeRDP/work/FreeRDP-1.1.0-beta+2013071101
```

This is a section of the macports log that shows the failure:
```
:info:configure -- FreeRDP GIT
:info:configure -- Git Revision n/a
:info:configure -- Finding optional feature X11 for X11 (X11 client and server)
:info:configure -- Skipping disabled feature DirectFB for DirectFB (DirectFB client)
:info:configure -- Finding required feature ZLIB for compression (data compression)
:info:configure -- Finding required feature OpenSSL for cryptography (encryption, certificate validation, hashing functions)
:info:configure -- Skipping recommended feature ALSA for sound (audio input, audio output and multimedia redirection)
:info:configure -- Skipping optional feature Pulse for sound (audio input, audio output and multimedia redirection)
:info:configure --     Enable feature Pulse using "-DWITH_PULSE=ON"
:info:configure -- Skipping optional feature Cups for printing (printer device redirection)
:info:configure --     Enable feature Cups using "-DWITH_CUPS=ON"
:info:configure -- Skipping optional feature PCSC for smart card (smart card device redirection)
:info:configure --     Enable feature PCSC using "-DWITH_PCSC=ON"
:info:configure -- Finding optional feature FFmpeg for multimedia (multimedia redirection, audio and video playback)
:info:configure -- Skipping optional feature Gstreamer for multimedia (multimedia redirection, audio and video playback)
:info:configure --     Enable feature Gstreamer using "-DWITH_GSTREAMER=ON"
:info:configure -- Skipping optional feature NPP for performance (NVIDIA Performance Primitives library)
:info:configure --     Enable feature NPP using "-DWITH_NPP=ON"
:info:configure -- FreeRDP GIT
:info:configure -- Finding recommended feature XKBFile for X11 keyboard (X11 keyboard file extension)
:info:configure --     Disable feature XKBFile using "-DWITH_XKBFILE=OFF"
:info:configure CMake Error at channels/client/CMakeLists.txt:33 (list):
:info:configure   list sub-command REMOVE_DUPLICATES requires list to be present.
:info:configure 
:info:configure 
:info:configure CMake Warning at client/X11/CMakeLists.txt:78 (message):
:info:configure   WITH_MANPAGES was set, but xmlto was not found.  man-pages will not be
:info:configure   installed
:info:configure 
:info:configure 
:info:configure -- Finding required feature XShm for X11 shared memory (X11 shared memory extension)
:info:configure -- Finding recommended feature Xinerama for multi-monitor (X11 multi-monitor extension)
:info:configure --     Disable feature Xinerama using "-DWITH_XINERAMA=OFF"
:info:configure -- Finding recommended feature Xext for X11 extension (X11 core extensions)
:info:configure --     Disable feature Xext using "-DWITH_XEXT=OFF"
:info:configure -- Finding recommended feature Xcursor for cursor (X11 cursor extension)
:info:configure --     Disable feature Xcursor using "-DWITH_XCURSOR=OFF"
:info:configure -- Finding recommended feature Xv for video (X11 video extension)
:info:configure --     Disable feature Xv using "-DWITH_XV=OFF"
:info:configure -- Finding recommended feature Xi for input (X11 input extension)
:info:configure --     Disable feature Xi using "-DWITH_XI=OFF"
:info:configure -- Finding recommended feature Xrender for rendering (X11 render extension)
:info:configure --     Disable feature Xrender using "-DWITH_XRENDER=OFF"
:info:configure -- Adding post-build NIB file generation event for MacFreeRDP-library
:info:configure -- Adding post-build NIB file generation event for MacFreeRDP
:info:configure -- Configuring incomplete, errors occurred!
:info:configure See also "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_net_FreeRDP/FreeRDP/work/FreeRDP-1.1.0-beta+2013071101/CMakeFiles/CMakeOutput.log".
:info:configure See also "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_net_FreeRDP/FreeRDP/work/FreeRDP-1.1.0-beta+2013071101/CMakeFiles/CMakeError.log".
:info:configure Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_net_FreeRDP/FreeRDP/work/FreeRDP-1.1.0-beta+2013071101" && /opt/local/bin/cmake -DCMAKE_INSTALL_PREFIX=/opt/local -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_INSTALL_RPATH=/opt/local/lib -DCMAKE_INSTALL_NAME_DIR=/opt/local/lib -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/usr" -DCMAKE_MODULE_PATH=/opt/local/share/cmake/Modules -DCMAKE_FIND_FRAMEWORK=LAST -Wno-dev -DWITH_X11=ON -DWITH_FFMPEG=ON -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.10" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_net_FreeRDP/FreeRDP/work/FreeRDP-1.1.0-beta+2013071101 
:info:configure Exit code: 1
```

**Comment by bmiklautz** [comment]
@diwu1989 the problem you are reporting is related to builds with cmake >= 3.0. This is fixed on master and stable-1.1 with commit (1b663ceffe51008af7ae9749e5b7999b2f7d6698 and b06549289081e73f92ef3fab593237340e9677aa respectively).

**Comment by diwu1989** [comment]
Ah cool, great, thank you
